sdk/appconfig/azappconfig/README.md

```diff
@@ -0,0 +1,305 @@
+    connectionString := os.Getenv("APPCONFIGURATION_CONNECTION_STRING")
+    client, err = azappconfig.NewClientFromConnectionString(connectionString, nil)
+    if err != nil {
+        panic(err)
+    }
+
+    // Get configuration setting
+    key := "key"
+    label := "label"
+    resp, err := client.GetSetting(context.TODO(), Setting{Key: &key, Label: &label}, nil)
+    if err != nil {
+        panic(err)
+    }
+    fmt.Println(*resp.Key)
+    fmt.Println(*resp.Label)
+    fmt.Println(*resp.Value)
+}
+```
+
+### Set a configuration setting
+
+```go
+import (
+    "fmt"
+    "os"
+
+    "github.com/Azure/azure-sdk-for-go/sdk/appconfig/azappconfig"
+)
+
+func ExampleSetConfigurationSetting() {
+    connectionString := os.Getenv("APPCONFIGURATION_CONNECTION_STRING")
+    client, err = azappconfig.NewClientFromConnectionString(connectionString, nil)
+    if err != nil {
+        panic(err)
+    }
+
+    // Set configuration setting
+    key := "key"
+    label := "label"
+    value := "new_value"
+    resp, err := client.SetSetting(context.TODO(), Setting{Key: &key, Label: &label, Value: &value}, nil)
+    if err != nil {
+        panic(err)
+    }
+    fmt.Println(*resp.Key)
+    fmt.Println(*resp.Label)
+    fmt.Println(*resp.Value)
+}
+```
+
+### Set a configuration setting read only
+
+```go
+import (
+    "fmt"
+    "os"
+
+    "github.com/Azure/azure-sdk-for-go/sdk/appconfig/azappconfig"
+)
+
+func ExampleSetConfigurationSettingReadOnly() {
+    connectionString := os.Getenv("APPCONFIGURATION_CONNECTION_STRING")
+    client, err = azappconfig.NewClientFromConnectionString(connectionString, nil)
+    if err != nil {
+        panic(err)
+    }
+
+    // Set configuration setting read only
+    key := "key"
+    label := "label"
+    resp, err := client.SetReadOnly(context.TODO(), Setting{Key: &key, Label: &label, true, nil)
+    if err != nil {
+        panic(err)
+    }
+    fmt.Println(*resp.Key)
+    fmt.Println(*resp.Label)
+    fmt.Println(*resp.Value)
+    fmt.Println(*resp.IsReadOnly)
+
+    // Remove read only status
+    resp, err := client.SetReadOnly(context.TODO(), Setting{Key: &key, Label: &label, false, nil)
+    if err != nil {
+        panic(err)
+    }
+    fmt.Println(*resp.Key)
+    fmt.Println(*resp.Label)
+    fmt.Println(*resp.Value)
+    fmt.Println(*resp.IsReadOnly)
+}
+```
+
+### List configuration setting revisions
+
+```go
+import (
+    "fmt"
+    "os"
+
+    "github.com/Azure/azure-sdk-for-go/sdk/appconfig/azappconfig"
+)
+
+func ExampleListRevisions() {
+    connectionString := os.Getenv("APPCONFIGURATION_CONNECTION_STRING")
+    client, err = azappconfig.NewClientFromConnectionString(connectionString, nil)
+    if err != nil {
+        panic(err)
+    }
+
+    any := "*"
+    revPgr := client.ListRevisions(SettingSelector{KeyFilter: &any, LabelFilter: &any, Fields: AllSettingFields()}, nil)
+    for revPgr.NextPage(context.TODO()) {
+        resp := revPgr.PageResponse()
```

Comment:
Which package is using the new API so I can use it as example? I'd change.

Reply:
I modernized the pager. But I think in the long term we should update codegen, so that it is easier to implement and so that it does not require to expose internal members.